charts/redpanda/chart_template_test.go

```diff
@@ -118,6 +123,37 @@ func TestTemplate(t *testing.T) {
 	}
 }
 
+func AssertAllVolumesAreUsed(t *testing.T, manifests []byte) {
```

Comment:
Is this the same assertion as https://github.com/redpanda-data/helm-charts/pull/1466/files#diff-9816759f5552bf9796d055b4dd5bb21fe21633582b33349ba828ac5d4092ffe2R126 ?

It isn't technically an error to have an extra volume that's not referenced in a volume mount (though it is likely bad practice :P )

Reply:
> It isn't technically an error to have an extra volume that's not referenced in a volume mount (though it is likely bad practice :P )

Yes, I agree that it's not an error.

> Is this the same assertion as https://github.com/redpanda-data/helm-charts/pull/1466/files#diff-9816759f5552bf9796d055b4dd5bb21fe21633582b33349ba828ac5d4092ffe2R126 ?

That assertion is other way around. In this change we are making sure that all volumes has at least one corresponding volume mount.

In the other PR it is all volume mounts have at least one corresponding volume.

> I also don't understand why you say this is a breaking change. Wouldn't this just be a small "bug" fix?

It is breaking change as StatefulSet does not support changes to the volumes. 

I will bump redpanda chart version to show you it will error out during chart testing execution.